### ISSUE TITLE
act-runner-task-executor container build fails

### ISSUE THREAD

**Issue by dboreham** [cerc-io/hosting#55]
Initially it looked like a stale base container issue, but further investigation suggests not.

```
$ laconic-so build-containers --include cerc/act-runner-task-executor
Dev Root is: /home/david/cerc
Building: cerc/act-runner-task-executor
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            BuildKit is currently disabled; enable it by removing the DOCKER_BUILDKIT=0
            environment-variable.

Sending build context to Docker daemon  3.119kB
Step 1/6 : FROM ubuntu:22.04
 ---> 5a81c4b8502e
Step 2/6 : RUN apt update && apt install -y gpg curl apt-transport-https ca-certificates lsb-release build-essential
 ---> Using cache
 ---> 53cb457e7718
Step 3/6 : RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
 ---> Running in 3404f4d42301
curl: (60) SSL: no alternative certificate subject name matches target host name 'download.docker.com'
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
gpg: no valid OpenPGP data found.
The command '/bin/sh -c curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg' returned a non-zero code: 2
Error running build for cerc/act-runner-task-executor
FATAL Error: container build failed and --continue-on-error not set, exiting
```

Running the curl command manually inside a freshly pulled ubuntu:22.04 container, I get the same error:

```
root@624223c046dc:/# curl  -vI https://download.docker.com/linux/ubuntu/gpg
*   Trying 18.160.143.28:443...
* Connected to download.docker.com (18.160.143.28) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS header, Finished (20):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.2 (OUT), TLS header, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.docker.com
*  start date: Feb 22 00:00:00 2023 GMT
*  expire date: Nov 30 23:59:59 2023 GMT
*  subjectAltName does not match download.docker.com
* SSL: no alternative certificate subject name matches target host name 'download.docker.com'
* Closing connection 0
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.3 (OUT), TLS alert, close notify (256):
curl: (60) SSL: no alternative certificate subject name matches target host name 'download.docker.com'
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

**Comment by dboreham** [comment]
Although curl says this:

```
SSL: no alternative certificate subject name matches target host name 'download.docker.com'
```

When I run it on the host it says:

```
subjectAltName: host "download.docker.com" matched cert's "*.docker.com"
```

This makes me think that the problem is actually that the CA cert is not available in the container.

**Comment by telackey** [comment]
That's part of what this line is for:

```
RUN apt update && apt install -y gpg curl apt-transport-https ca-certificates lsb-release build-essential
```

Your output has:

```
Step 2/6 : RUN apt update && apt install -y gpg curl apt-transport-https ca-certificates lsb-release build-essential
 ---> Using cache
 ---> 53cb457e7718
```

Perhaps clear 53cb457e7718?

**Comment by dboreham** [comment]
That cached layer was from a previous `--no-cache` build.
But let me try again just to be sure...

**Comment by dboreham** [comment]
Works for me now.
